fluent-syntax/src/parser.js

```diff
@@ -383,22 +380,24 @@ export default class FluentParser {
   }
 
   getNumber(ps) {
-    let num = "";
+    let integer = "";
+    let fraction = "";
 
     if (ps.currentChar === "-") {
-      num += "-";
       ps.next();
+      integer = `-${this.getDigits(ps)}`;
+    } else {
+      integer = this.getDigits(ps);
     }
 
-    num = `${num}${this.getDigits(ps)}`;
-
     if (ps.currentChar === ".") {
-      num += ".";
       ps.next();
-      num = `${num}${this.getDigits(ps)}`;
+      fraction = this.getDigits(ps);
     }
 
-    return new AST.NumberLiteral(num);
+    let value = parseFloat(`${integer}.${fraction}`);
```

Comment:
I like to think about AST as a structured, semantic view of the source string.

From that perspective, what if we stored a number as a slice `[start;end]` and parsed it in the resolver instead?

My concern here is that storing numbers and parsing them differs between different programming languages Fluent may work in. Number is a low level primitive, but it still can differ in how it stores fractions, unsigned vs signed etc. By keeping AST free of that logic, we ensure that the AST is portable between the implementations irrelevant of how JS, Python or Rust treat numerical values.

Not a blocker, just a suggestion.

Reply:
To close the loop: this was fixed in the spec in https://github.com/projectfluent/fluent/issues/239. I've updated the PR to reflect it.